query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLQuery.java

```diff
@@ -0,0 +1,39 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder(builderClassName = "Builder", buildMethodName = "build")
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class PromQLQuery {
+
+  @NonNull @Singular private List<String> queries;
+  private boolean isInstantRequest;
+  private Instant startTime;
+  private Instant endTime;
+  private Duration step;
+
+  static class Builder {
+    PromQLQuery build() {
+      if (queries.size() == 0) {
+        throw new UnsupportedOperationException(
+            "query string requires for Instant or Range query request");
```

Comment:
nit : requires -> required

Reply:
No need. Broken PromQLQuery objects into PromQLInstantQuery and PromQLRangeQuery.